src/benefits/benefits.service.ts

```
@@ -287,20 +285,20 @@ export class BenefitsService {
       }
 
       // Generate order ID
-      const orderId: string = benefit?.orderId ? benefit.orderId : `TLEXP_${this.generateRandomString()}_${Date.now()}`;  
+      const orderId: string = benefit?.orderId ?? `TLEXP_${this.generateRandomString()}_${Date.now()}`;
 
       // Update customer details
       const orderDetails = await this.applicationsService.update(Number(applicationId), { orderId });
 
-      const { id, descriptor, categories, locations, items, rateable }: any =
+      const { id, descriptor, locations, items, rateable }: any =
         mappedResponse?.message.catalog.providers[0];
```

Biome (1.9.4)

[error] Unsafe usage of optional chaining.

If it short-circuits with 'undefined' the evaluation will throw TypeError here:


(lint/correctness/noUnsafeOptionalChaining)